src/utils.ts

```diff
@@ -101,10 +101,7 @@ export class Utils {
                 } else {
                     const name = var1 || var2;
                     assert(name, "unexpected unset capture group");
-                    let value = `${expandWith.variable(name)}`;
-                    if (value.startsWith("\"/") && value.endsWith("/\"")) {
-                        value = value.substring(1).slice(0, -1);
-                    }
```

Comment:
Removed this chunk because, 

in order to support the following:

```yaml
job:
  image: alpine
  variables:
    teststring: "/test/url"
    pattern: "/test//"
  script:
    - echo works
  rules:
    - if: '$teststring =~ $pattern' # works
    - if: $teststring =~ /test//    # does not work
```